third_party/tlx/language/mma_ops.py

```diff
@@ -6,37 +6,34 @@
 from .utility import cuda_parse_arch
 
 
-def require_nv_mma_shared_layout(x: tlx.buffered_tensor, order, _builder=None):
-    if not isinstance(x.layout, tlx.nv_mma_shared_layout_encoding):
-        # TODO. why do we need this class object?
-        layout = tlx.nv_mma_shared_layout_encoding(shape=x.shape, order=order, elemType=x.dtype, numCTAsPerCGA=[1, 1],
+def require_nv_mma_shared_layout(x: tlx.buffered_tensor, _builder=None):
+    assert isinstance(x.layout, tlx.shared_layout_encoding), "input must be a shared tensor"
+    if isinstance(x.layout, tlx.swizzled_shared_layout_encoding):
+        layout = tlx.nv_mma_shared_layout_encoding(shape=x.shape, order=x.layout.order, elemType=x.dtype, numCTAsPerCGA=[1, 1],
                                                    numCTASplit=[1, 1], numCTAOrder=[1, 1], fp4Padded=False)
-
-    layout_handle = _builder.make_nv_mma_shared_encoding_attr(
-        [int(x) for x in layout.shape],
-        layout.order,
-        layout.elemType.to_ir(_builder),
-        layout.numCTAsPerCGA,
-        layout.numCTASplit,
-        layout.numCTAOrder,
-        layout.fp4Padded,
-    )
-    return _builder.create_require_layout(x.handle, layout_handle)
+        layout_handle = _builder.make_nv_mma_shared_encoding_attr(
+            [int(x) for x in layout.shape],
+            layout.order,
+            layout.elemType.to_ir(_builder),
+            layout.numCTAsPerCGA,
+            layout.numCTASplit,
+            layout.numCTAOrder,
+            layout.fp4Padded,
+        )
+        return _builder.create_require_layout(x.handle, layout_handle)
+    else:
+        assert isinstance(x.layout, tlx.nv_mma_shared_layout_encoding), "input must be a shared mma tensor"
+        return x.handle
 
 
 # async dot signature needs to be close to tl.dot as much as possible
 @tl.builtin
 def async_dot(
-    input: tlx.buffered_tensor,
-    other: tlx.buffered_tensor,
-    acc=None,  # tl.tensor,
-    mmav5: bool = False,
```

Comment:
@pchen7e2 I'm removing the mmav5 flag, instead computing in on the fly. Please let me know if you have any concerns.